crates/lib3h/src/engine/ghost_engine_wrapper.rs

```diff
@@ -281,22 +301,32 @@ where
     pub fn name(&self) -> String {
         self.name.clone()
     }
-}
 
-fn inject_request_id(request_id: String, mut msg: Lib3hServerProtocol) -> Lib3hServerProtocol {
-    match &mut msg {
-        Lib3hServerProtocol::Connected(data) => data.request_id = request_id,
-        Lib3hServerProtocol::FetchEntryResult(data) => data.request_id = request_id,
-        Lib3hServerProtocol::HandleSendDirectMessage(data) => data.request_id = request_id,
-        Lib3hServerProtocol::HandleFetchEntry(data) => data.request_id = request_id,
-        Lib3hServerProtocol::HandleStoreEntryAspect(data) => data.request_id = request_id,
-        Lib3hServerProtocol::HandleDropEntry(data) => data.request_id = request_id,
-        Lib3hServerProtocol::HandleQueryEntry(data) => data.request_id = request_id,
-        Lib3hServerProtocol::HandleGetAuthoringEntryList(data) => data.request_id = request_id,
-        Lib3hServerProtocol::HandleGetGossipingEntryList(data) => data.request_id = request_id,
-        _ => (),
+    fn inject_request_id(
+        &mut self,
+        request_id: String,
+        mut msg: Lib3hServerProtocol,
+    ) -> Lib3hServerProtocol {
+        match &mut msg {
+            Lib3hServerProtocol::Connected(data) => data.request_id = request_id,
+            Lib3hServerProtocol::FetchEntryResult(data) => data.request_id = request_id,
+            Lib3hServerProtocol::HandleFetchEntry(data) => data.request_id = request_id,
+            Lib3hServerProtocol::HandleStoreEntryAspect(data) => data.request_id = request_id,
+            Lib3hServerProtocol::HandleDropEntry(data) => data.request_id = request_id,
+            Lib3hServerProtocol::HandleQueryEntry(data) => data.request_id = request_id,
+            Lib3hServerProtocol::HandleGetAuthoringEntryList(data) => data.request_id = request_id,
+            Lib3hServerProtocol::HandleGetGossipingEntryList(data) => data.request_id = request_id,
+            Lib3hServerProtocol::HandleSendDirectMessage(data) => {
+                // Cache this request_id as the ghost engine layer needs it to associate
+                // the response message
```

Comment:
This is the part where I'm pretty sure something lower is relying on a request_id that shouldn't be... The lower level received a request_id over the P2p message, and should be capturing it in the callback when it makes the request up to here, and shouldn't be paying attention to the one we re-inject from the request_id_map above.

I'm totally cool with merging this, especially if it means we have a working test, so we can muck with the lower code and prove it still works :+1:

Reply:
Agreed, would like to get rid of this request_id dependency!